File: src/main/java/com/netcracker/models/DebtPaymentOperation.java
package com.netcracker.models;

import lombok.Data;
import lombok.EqualsAndHashCode;

import java.math.BigInteger;
import java.util.Date;

@EqualsAndHashCode(callSuper = true)
@Data
public class DebtPaymentOperation extends AbstractOperation {
    private ApartmentSubBill apartmentSubBill;
    private ManagerSubBill managerSubBill;

    public DebtPaymentOperation(BigInteger debtPaymentOperation, Double sum, Date createdAt, ApartmentSubBill apartmentSubBill, ManagerSubBill managerSubBill){
        super(debtPaymentOperation, sum, createdAt);
        this.apartmentSubBill = apartmentSubBill;
        this.managerSubBill = managerSubBill;
    }
}

File: src/main/angular/app/models/debt-payment-operation.ts
import {ApartmentSubBill} from "./apartment-sub-bill";
import {ManagerSubBill} from "./manager-sub-bill";

export class DebtPaymentOperation{
    operationId?: number;
    sum?: number;
    createdAt?: Date;
    apartmentSubBill?: ApartmentSubBill;
    managerSubBill?: ManagerSubBill;
}